examples/server/server.cpp

```diff
@@ -3028,13 +3028,25 @@ int main(int argc, char ** argv) {
         log_data["api_key"] = "api_key: " + std::to_string(sparams.api_keys.size()) + " keys loaded";
     }
 
-    // load the model
-    if (!ctx_server.load_model(params)) {
-        state.store(SERVER_STATE_ERROR);
-        return 1;
-    } else {
+    try {
+        // Attempt to load the model
+        if (!ctx_server.load_model(params)) {
+            LOG_ERROR("unable to load model", {{"error", "Load model failed"}});
+            state.store(SERVER_STATE_ERROR);
+            return 1;
+        }
+        // Initialize the server context if the model is loaded successfully
         ctx_server.init();
         state.store(SERVER_STATE_READY);
+
+    } catch (const std::exception &e) {
+        // Catch known standard exceptions
+        LOG_ERROR("unable to load model", {{"error", e.what()}});
+        exit(1);
```

Comment:
You forgot to change this.

Reply:
ok.. I give up... 3 days and still nothing for a simple modification..
also: on windows, the program crashes for every stupid error like putting a non existant model filename.
that just plain wrong.
all exceptions should be caught. mine was only an example.

Reply:
![image](https://github.com/ggerganov/llama.cpp/assets/170285982/ea7653e1-f9e1-4d1d-8e1d-1464288cfb54)